shared/middleware/files_api.rb

```diff
@@ -374,11 +374,11 @@ def copy_file(endpoint, encrypted_channel_id, filename, source_filename)
   end
 
   #
-  # PUT /v3/sources/<channel-id>/<filename>?version=<version-id>
+  # PUT /v3/(sources|assets)/<channel-id>/<filename>?version=<version-id>
```

Comment:
Passing a `version` parameter with to the assets API might not go well - I'd consider breaking this comment into two lines, so the assets case clearly doesn't support the version parameter, and also adding a unit test for trying to pass a version when uploading an asset.  I think the ideal behavior would be to just ignore the parameter in that case.